src/proto/NULS.proto

```diff
@@ -0,0 +1,66 @@
+syntax = "proto3";
+
+package TW.NULS.Proto;
+option java_package = "wallet.core.jni.proto";
+
+message TransactionCoinFrom {
+    string from_address = 1;
+    uint32 assets_chainid = 2;
+    uint32 idassets_id = 3;
+    //tranaction out amount (128-bit number)
+    bytes idamount = 4;
+    //8 bytes
+    bytes nonce = 5;
+    //lock status: 1 locked; 0 unlocked
+    uint32 locked = 6;
+}
+
+message TransactionCoinTo {
+    string to_address = 1;
+    uint32 assets_chainid = 2;
+    uint32 idassets_id = 3;
+    // tranaction amount (128-bit number)
+    bytes idamount = 4;
+    uint32 lock_time = 5;
+}
+
+message Signature {
+    uint32 pkey_len = 1;
+    bytes public_key = 2;
+    uint32 sig_len = 3;
+    bytes signature = 4;
+}
+
+message Transaction {
+    uint32  type = 1;
+    uint32  timestamp = 2;
+    string  remark = 3;
+    bytes   tx_data = 4;
+    //CoinFrom
+    repeated TransactionCoinFrom inputs = 5;
+    //CoinTo
+    repeated TransactionCoinTo outputs = 6;
+
+    repeated Signature tx_sigs = 7;
+    uint32 hash = 8;
+}
+
+// Input data necessary to create a signed order.
+message SigningInput {
+    bytes private_key = 1;
+    string from_address = 2;
```

Comment:
you can get from address from private key, so you don't have to pass it

Reply:
What is the api to calll to get private key from address?
I think the code would run in wallet core of  TW APP.

Reply:
you can derive address from your private key, that should be your `from_address`?

Reply:
I want to get private key to signature transaction.
How could  I get private key from from_address which is carried in protobuf message?

Reply:
@hewigovens  please reply how to getting private key. I read token proto files defining private key attribute.

Reply:
do you mean this?
```c++
// Proto::SigningInput &input
PrivateKey(Data(input.private_key().begin(), input.private_key().end()));
```

Reply:
Yes, how to get private key from proto message or other ways.